tensorflow_probability/python/experimental/mcmc/sequential_monte_carlo_kernel.py

```diff
@@ -145,6 +148,7 @@ def __init__(self,
                resample_fn=weighted_resampling.resample_systematic,
                resample_criterion_fn=ess_below_threshold,
                unbiased_gradients=True,
+               particles_dim=0,
```

Comment:
It looks like the `bootstrap` method also needs to be updated to handle `particles_dim`.

That is, instead of initializing `incremental_log_marginal_likelihood` and `accumulated_log_marginal_likelihood` to:
```
batch_zeros = tf.zeros(
    ps.shape(init_state.log_weights)[1:],
    dtype=init_state.log_weights.dtype)
```
we need to initialize them to something like:
```
        particles_shape = ps.shape(init_state.log_weights)
        weights_shape = ps.concat([
            particles_shape[:self.particles_dim],
            particles_shape[self.particles_dim+1:]
        ], axis=0)
        batch_zeros = tf.zeros(
            weights_shape, dtype=init_state.log_weights.dtype)
```


To verify that we have done this correctly, could you also please add a version of `test_batch_of_filters` in `particle_filter_test.py` with a `particles_dim` of 1 or 2?